src/LightingThread.cpp

```diff
@@ -357,7 +357,7 @@ void cLightingThread::PrepareSkyLight(void)
 	m_NumSeeds = 0;
 
 	// Fill the top of the chunk with all-light:
-	if (m_MaxHeight < cChunkDef::Height - 1)
+	if (cChunkDef::IsValidHeight(m_MaxHeight, 1))
```

Comment:
:exclamation: This needs rework, `m_MaxHeight` is used as an index into the internal array, so must be always 0-based, unlike an Y-coord.